streaming/base/compression/compression.py

```diff
@@ -132,7 +132,7 @@ class Snappy(Compression):
 
     extension = 'snappy'
 
-    def compress(self, data: bytes) -> bytes:
+    def compress(self, data: Union[bytes, str]) -> bytes:
```

Comment:
Looks like snappy compression algorithms takes string as well apart from bytes. Hence, made this changes.

Reply:
we are not using this with strings for our purposes, let's not enable that use case here?

Reply:
Looks like python annotations is very strict. If we keep it as `bytes` and you pass a string, then it won't throw any error and at a same time, it compressed the data and return the data in `bytes` !! 😞

Reply:
would rather keep it uniform with the other compression algo interfaces and call this silent extra capability an easter egg? up to you if you feel strongly about it

Reply:
Cool. Let me change it back to `bytes` as this is not a major concern.

Reply:
Done in https://github.com/mosaicml/streaming/pull/25/commits/0e5d87e61ee2b027df73751826c615755ffdf499.